src/platform/linux/x11/mod.rs

```diff
@@ -499,20 +499,24 @@ impl EventsLoop {
                     }
 
                     // This is a hack to ensure that the DPI adjusted resize is actually applied on all WMs. KWin
-                    // doesn't need this, but Xfwm does.
+                    // doesn't need this, but Xfwm does. The hack should not be run on other WMs, since tiling
+                    // WMs constrain the window size, making the resize fail. This would cause an endless stream of
+                    // XResizeWindow requests, making Xorg, the winit client, and the WM consume 100% of CPU.
                     if let Some(adjusted_size) = shared_state_lock.dpi_adjusted {
-                        let rounded_size = (adjusted_size.0.round() as u32, adjusted_size.1.round() as u32);
-                        if new_inner_size == rounded_size {
-                            // When this finally happens, the event will not be synthetic.
-                            shared_state_lock.dpi_adjusted = None;
-                        } else {
-                            unsafe {
-                                (self.xconn.xlib.XResizeWindow)(
-                                    self.xconn.display,
-                                    xwindow,
-                                    rounded_size.0 as c_uint,
-                                    rounded_size.1 as c_uint,
-                                );
+                        if util::wm_name_is_one_of(&["Xfwm4"]) {
```

Comment:
I wonder if this should be combined with the `new_inner_size == rounded_size` two lines below? Any opinions?

Reply:
I realized that with the check here the WM name comparison would be done on every `XConfigureWindow`. I combined the check with the one below and force-pushed. I re-tested with Xfwm4 and i3.